apps/frontend/src/pages/overview/libs/components/add-course-modal/add-course-modal.tsx

```diff
@@ -0,0 +1,131 @@
+import { Courses, Input, Loader, Modal } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import { debounce } from "~/libs/helpers/helpers.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+} from "~/libs/hooks/hooks.js";
+import { actions as courseActions } from "~/modules/courses/courses.js";
+import {
+	type AddCourseRequestDto,
+	actions as userCourseActions,
+} from "~/modules/user-courses/user-courses.js";
+import { actions as vendorActions } from "~/modules/vendors/vendors.js";
+
+import {
+	DEFAULT_SEARCH_COURSE_PAYLOAD,
+	SEARCH_COURSES_DELAY_MS,
+} from "../../constants/constants.js";
+import {
+	getDefaultVendors,
+	getVendorsFromForm,
+} from "../../helpers/helpers.js";
+import { VendorBadge } from "../vendor-badge/vendor-badge.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	onClose: () => void;
+};
+
+const AddCourseModal: React.FC<Properties> = ({ onClose }: Properties) => {
+	const dispatch = useAppDispatch();
+	const { courses, isLoading, vendors } = useAppSelector((state) => {
+		return {
+			courses: state.courses.searchedCourses,
+			isLoading: state.courses.searchDataStatus === DataStatus.PENDING,
+			vendors: state.vendors.vendors,
+		};
+	});
+	const { control, errors, getValues, setValue } = useAppForm({
+		defaultValues: DEFAULT_SEARCH_COURSE_PAYLOAD,
+		mode: "onChange",
+	});
+
+	const handleAddCourse = useCallback(
+		(payload: AddCourseRequestDto) => {
+			void dispatch(userCourseActions.add(payload));
+		},
+		[dispatch],
+	);
+
+	const handleSearchCourses = (): void => {
+		const formData = getValues();
```

Comment:
```suggestion
	const handleSearchCourses = (data: Type): void => {
```
And use `handleSubmit` from `useAppForm`